lib/output.js

```diff
@@ -1963,19 +1981,28 @@ function OutputStream(options) {
         return best;
     }
 
-    function make_num(num) {
-        var str = num.toString(10), a = [ str.replace(/^0\./, ".").replace("e+", "e") ], m;
+    function make_num(num, bigInt = false) {
+        var str = num.toString(10).replace(/^0\./, ".").replace("e+", "e");
+        var candidates = [ str ];
         if (Math.floor(num) === num) {
```

Comment:
This won't work if we change `AST_BigInt.value` to be a `BigInt`, since `Math.floor()` does not work on `BigInt` inputs.

We'll want this check to always pass if `num` is a `BigInt`. (Or if `bigInt` is `true`, but I think that should be the same?)

Reply:
Or in case if we will store BigInt value as a string than I propose to keep it as is